d8/modules/custom/dmaps/src/Form/DmapsGeocoderOptionsForm.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Contains \Drupal\dmaps\Form\DmapsGeocoderOptionsForm
+ */
+
+namespace Drupal\dmaps\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+
+class DmapsGeocoderOptionsForm extends ConfigFormBase {
+  /**
+   * Country iso
+   *
+   * @var mixed
+   */
+  protected $iso;
+  /**
+   * Geocoder provider name
+   *
+   * @var mixed
+   */
+  protected $service;
+
+  public function __construct(\Drupal\Core\Config\ConfigFactoryInterface $config_factory) {
+    $request = \Drupal::request();
+    $this->iso = $request->attributes->get('iso');
+    $this->service = $request->attributes->get('service');
+
+    parent::__construct($config_factory);
+  }
+
+  public function getFormId() {
+    return 'dmaps_geocoder_options';
+  }
+
+  public function getEditableConfigNames() {
+    return ['dmaps.geocoder_' . $this->iso . '_' . $this->service . '_settings'];
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config_name = 'dmaps.geocoder.' . $this->service . '_' . $this->iso;
+    $config = \Drupal::configFactory()->getEditable($config_name);
+    $country_manager = \Drupal::service('dmaps.location_countries_manager');
+    $country_manager->locationLoadCountry($this->iso);
+
+    // @todo Need to refactor when geocoder plugins will be implemented
+    $geocode_settings_form_function_specific = 'location_geocode_' . $this->iso . '_' . $this->service . '_settings';
+    $geocode_settings_form_function_general = $this->service . '_geocode_settings';
+    if (function_exists($geocode_settings_form_function_specific)) {
+      $form = $geocode_settings_form_function_specific($this->iso, $config);
+      parent::buildForm($form, $form_state);
+    }
+
+    $geocoder = \Drupal::service('dmaps.geocoder');
```

Comment:
Add both services `dmaps.location_countries_manager` and `dmaps.geocoder` to the form constructor. 

Example for language_manager: 
```php
  /**
   * {@inheritdoc}
   */
  public static function create(ContainerInterface $container) {
    return new static(
      $container->get('config.factory'),
      $container->get('language_manager')
    );
  }
  /**
   * Constructs a \Drupal\image_captcha\Form\ImageCaptchaSettingsForm object.
   *
   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
   *   The factory for configuration objects.
   * @param \Drupal\Core\Language\LanguageManagerInterface $language_manager
   *   The language manager.
   */
  public function __construct(ConfigFactoryInterface $config_factory, LanguageManagerInterface $language_manager) {
    parent::__construct($config_factory);
    $this->languageManager = $language_manager;
  }
```